pkg/virt-launcher/virtwrap/nichotplug.go

```diff
@@ -32,31 +32,18 @@ import (
 	netvmispec "kubevirt.io/kubevirt/pkg/network/vmispec"
 	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
 	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/cli"
-	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/converter"
 )
 
-func hotplugVirtioInterface(vmi *v1.VirtualMachineInstance, converterContext *converter.ConverterContext, dom cli.VirDomain, domain *api.Domain) error {
+func hotplugVirtioInterface(vmi *v1.VirtualMachineInstance, dom cli.VirDomain, currentDomain *api.Domain, updatedDomain *api.Domain) error {
 	vmConfigurator := netsetup.NewVMNetworkConfigurator(vmi, cache.CacheCreator{})
-	indexedInterfaces := netvmispec.IndexInterfaceSpecByName(vmi.Spec.Domain.Devices.Interfaces)
-	for _, network := range netvmispec.NetworksToHotplugWhosePodIfacesAreReady(vmi) {
+	for _, network := range networksToHotplugWhoseInterfacesAreNotInTheDomain(vmi, indexedDomainInterfaces(currentDomain)) {
```

Comment:
It seems that `networksToHotplugWhoseInterfacesAreNotInTheDomain` returns a VMI `spec.network` slice to fit `SetupPodNetworkPhase2`, and then use that network object to get the domain interface object.

Eventually,  only the VMI spec.network / status.interface name (same thing) is needed to get the right VMI spec.network and the domain interface.

How about changing it to first find the names of the interfaces to hotplug, get the objects and act accordingly (phase2 / attchtDevice)?

I think it will simplify the code in a way it will be easier to follow and save some indexing operations.

Example:
```
for _, ifaceName := range ifaceNamesToHotplug(vmi.status.interfaces, domain.Spec.Devices.Interfaces) {
    network := networkByName(vmi.spec.networks, ifaceName)
    // perform pod network setup phase2
    ...

    domainIface := domainIfaceByName(domain.Spec.Devices.Interfaces, ifaceName)
    //perform devices attachment
    ...
}
```